sync/sync_test.go

```diff
@@ -86,9 +87,13 @@ func TestSyncBlocks(t *testing.T) {
 
 		mockSNData := mocks.NewMockStarknetData(mockCtrl)
 
+		var rwMutex sync.RWMutex
 		syncingHeight := uint64(0)
```

Comment:
Cool, using an atomic would be even better https://pkg.go.dev/sync/atomic#Int64

Reply:
Will explore this and see if it can be done in a separate PR.